src/test/java/com/commercetools/sync/products/utils/BuildProductVariantImagesUpdateActionsTest.java

```diff
@@ -0,0 +1,329 @@
+        final MoveImageToPosition moveImage4ToPosition = (MoveImageToPosition) updateActions.get(5);
+
+        assertThat(removeImage).isExactlyInstanceOf(RemoveImage.class);
+        assertThat(removeImage.getImageUrl()).isEqualTo(image.getUrl());
+
+        assertThat(addImage).isExactlyInstanceOf(AddExternalImage.class);
+        assertThat(addImage.getImage()).isEqualTo(image4);
+
+        assertThat(moveImage5ToPosition).isExactlyInstanceOf(MoveImageToPosition.class);
+        assertThat(moveImage5ToPosition.getImageUrl()).isEqualTo(image5.getUrl());
+        assertThat(moveImage5ToPosition.getPosition()).isEqualTo(newImages.indexOf(image5));
+
+        assertThat(moveImage2ToPosition).isExactlyInstanceOf(MoveImageToPosition.class);
+        assertThat(moveImage2ToPosition.getImageUrl()).isEqualTo(image2.getUrl());
+        assertThat(moveImage2ToPosition.getPosition()).isEqualTo(newImages.indexOf(image2));
+
+        assertThat(moveImage3ToPosition).isExactlyInstanceOf(MoveImageToPosition.class);
+        assertThat(moveImage3ToPosition.getImageUrl()).isEqualTo(image3.getUrl());
+        assertThat(moveImage3ToPosition.getPosition()).isEqualTo(newImages.indexOf(image3));
+
+        assertThat(moveImage4ToPosition).isExactlyInstanceOf(MoveImageToPosition.class);
+        assertThat(moveImage4ToPosition.getImageUrl()).isEqualTo(image4.getUrl());
+        assertThat(moveImage4ToPosition.getPosition()).isEqualTo(newImages.indexOf(image4));
+    }
+
+    @Test
+    public void buildMoveImageToPositionUpdateActions_WithDifferentOrder_ShouldBuildActions() {
+        final Image image2 = Image.of("https://image2.url.com", ImageDimensions.of(2, 2), "image2Label");
+        final Image image3 = Image.of("https://image3.url.com", ImageDimensions.of(2, 2), "image3Label");
+        final Image image4 = Image.of("https://image4.url.com", ImageDimensions.of(2, 2), "image4Label");
+        final Image image5 = Image.of("https://image5.url.com", ImageDimensions.of(2, 2), "image5Label");
+
+        final List<Image> oldImages = new ArrayList<>();
+        oldImages.add(image5);
+        oldImages.add(image2);
+        oldImages.add(image3);
+        oldImages.add(image4);
+
+        final List<Image> newImages = new ArrayList<>();
+        newImages.add(image4);
+        newImages.add(image5);
+        newImages.add(image2);
+        newImages.add(image3);
+
+        final List<MoveImageToPosition> updateActions =
+            buildMoveImageToPositionUpdateActions(1, oldImages, newImages);
+
+        assertThat(updateActions).containsExactlyInAnyOrder(
+            MoveImageToPosition.ofImageUrlAndVariantId(image4.getUrl(), 1, 0, true),
+            MoveImageToPosition.ofImageUrlAndVariantId(image5.getUrl(), 1, 1, true),
+            MoveImageToPosition.ofImageUrlAndVariantId(image2.getUrl(), 1, 2, true),
+            MoveImageToPosition.ofImageUrlAndVariantId(image3.getUrl(), 1, 3, true));
+    }
```

Comment:
note, i simplified the solution: use `ListAssert#containsExactlyInAnyOrder(ELEMENT... )` instead of 4 different asserts.

Reply:
thanks